### ISSUE TITLE
Whether the `apply` function is implemented using the pandas_udf function?

### ISSUE THREAD

**Issue by RainFung** [databricks/koalas#2217]
Whether the `apply` function is implemented using the pandas_udf function?

**Comment by itholic** [comment]
Yeah, for some case we use `mapInPandas` which leveraging the `pandas_udf` internally.

https://github.com/databricks/koalas/blob/07d846225d27492a9b85e58900018fda576650e3/databricks/koalas/frame.py#L2557-L2563
